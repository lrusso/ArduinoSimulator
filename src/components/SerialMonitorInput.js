import React from "react"

const SerialMonitorInput = ({
  sendSerialData,
  value,
  onChange,
  simulatorRunning,
}) => {
  const refInput = React.useRef(null)

  const _onKeyUp = (event) => {
    if (event.key === "Enter") {
      sendSerialData()
      refInput.current.focus()
    }
  }

  const _onChange = (event) => {
    onChange(event.target.value)
  }

  return (
    <input
      ref={refInput}
      type="text"
      value={value}
      onChange={_onChange}
      onKeyUp={_onKeyUp}
      readOnly={!simulatorRunning}
      style={simulatorRunning ? styles.enabled : styles.disabled}
    />
  )
}

const styles = {
  enabled: {
    flex: 2,
    borderRadius: 0,
    marginLeft: "5px",
    marginBottom: "7px",
    marginRight: "3px",
    border: "1px solid silver",
    outline: "none",
    fontFamily: "Arial",
    fontSize: "13px",
    height: "18px",
    padding: "4px",
    WebkitAppearance: "none",
  },
  disabled: {
    flex: 2,
    borderRadius: 0,
    marginLeft: "5px",
    marginBottom: "7px",
    marginRight: "3px",
    border: "1px solid silver",
    outline: "none",
    fontFamily: "Arial",
    fontSize: "13px",
    height: "18px",
    padding: "4px",
    cursor: "default",
    WebkitAppearance: "none",
    backgroundColor: "#D9D9D9",
  },
}

export default SerialMonitorInput
