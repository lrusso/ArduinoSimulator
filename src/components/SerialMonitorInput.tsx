import React from "react"

interface SerialMonitorInputProps {
  sendSerialData: () => void
  value: string
  // eslint-disable-next-line no-unused-vars
  onChange: (value: string) => void
  simulatorRunning: boolean
}

const SerialMonitorInput = ({
  sendSerialData,
  value,
  onChange,
  simulatorRunning,
}: SerialMonitorInputProps) => {
  const refInput = React.useRef<HTMLInputElement>(null)

  const _onKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === "Enter") {
      sendSerialData()
      refInput.current.focus()
    }
  }

  const _onChange = (event: { target: HTMLInputElement }) => {
    onChange(event.target.value)
  }

  return (
    <input
      ref={refInput}
      className="serialInput"
      type="text"
      value={value}
      onChange={_onChange}
      onKeyUp={_onKeyUp}
      readOnly={!simulatorRunning}
      disabled={!simulatorRunning}    
    />
  )
}

export default SerialMonitorInput
