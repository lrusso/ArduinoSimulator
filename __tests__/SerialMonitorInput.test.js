import React from "react"
import renderer from "react-test-renderer"
import SerialMonitorInput from "../src/components/SerialMonitorInput"

describe("SerialMonitorInput", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<SerialMonitorInput />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
