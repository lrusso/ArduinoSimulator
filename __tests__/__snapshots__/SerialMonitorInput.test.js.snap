// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SerialMonitorInput Rendering 1 1`] = `
<input
  onChange={[Function]}
  onKeyUp={[Function]}
  readOnly={true}
  style={
    {
      "WebkitAppearance": "none",
      "backgroundColor": "#D9D9D9",
      "border": "1px solid silver",
      "borderRadius": 0,
      "cursor": "default",
      "flex": 2,
      "fontFamily": "Arial",
      "fontSize": "13px",
      "height": "18px",
      "marginBottom": "7px",
      "marginLeft": "5px",
      "marginRight": "3px",
      "outline": "none",
      "padding": "4px",
    }
  }
  type="text"
/>
`;
